crates/types/src/blockchain/header.rs

```diff
@@ -436,13 +436,18 @@ impl ApplicationHeader<GeneratedApplicationFields> {
     pub fn hash(&self) -> Bytes32 {
         // Order matters and is the same as the spec.
         let mut hasher = crate::fuel_crypto::Hasher::default();
-        hasher.input(self.da_height.to_bytes().as_slice());
+        hasher.input(self.da_height.to_be_bytes().as_slice());
         hasher.input(self.transactions_count.to_be_bytes());
```

Comment:
@xgreenx said on Slack:
>The TxPointer.index_output is u16, so it means we can’t have more transactions than u16, but we use u64. The same for the messages receipts, we can’t have more than transactions_count * 255, so it can’t be more than u32=)
If we don’t need to pad them and it is a bug in the transactions_count and message_receipt_count, then we can remove it=)

But the docs still say:
```
| `txCount`                        | `uint64`   | Number of [transaction](../tx-format/transaction.md)s in this block.                                                                                                                                                                                |
| `message_receipt_count`          | `uint64`   | Number of [output message](../abi/receipts.md#messageout-receipt)s in this block.     
```
So I kept these to match the specs.

Reply:
https://github.com/FuelLabs/fuel-specs/pull/578